packages/serve/src/createConfig.ts

```diff
@@ -18,8 +22,7 @@ export default function createConfig(args): devServerOptionsType {
         // clientLogging is not a valid devServer option
         delete options.clientLogging;
     }
-
-    if (options.hotOnly) {
+    if (isDevServer4 && options.hotOnly) {
```

Comment:
I think we don't have `options.hotOnly` on CLI only `hot` and `hot = 'only'`

Reply:
we have it on v3

Reply:
But options from webpack-cli

Reply:
nope options here is the dev server flags

Reply:
printed and checked 😄

Reply:
thanks